tests/src/highlevel-promise.js

```diff
@@ -120,7 +120,7 @@ export class HighlevelPromiseContract {
     return {
       ...callingData(),
       promiseResults: arrayN(near.promiseResultsCount()).map((i) =>
-        near.promiseResult(i)
+        str(near.promiseResult(i))
```

Comment:
Any use cases we want to represent `promiseResult` as something else but string?
(this one is about moving conversion to the API level again)

Reply:
One example would be promise return one's public key.  And callback function take that public key to do something. Public keys that host function returns and takes are binary format.

Reply:
So, if they both have string in their API, it will work? But we will do the conversion 2 times (gas issue)?

Reply:
Yes. conversion 2 times adds unnecessary gas consumption